### PR TITLE
buttons flags cleanup

### LINKED ISSUE #7968: some custom widget "icon flags" can be superceded by CSS
### Is your feature request related to a problem? Please describe.
Custom icons `GtkDarktableButton`, `GtkDarktableToggleButton`, `GtkDarktableThumbnailBtn`, `GtkDarktableIcon` take `icon_flags`:

https://github.com/darktable-org/darktable/blob/ad87231986fa519d2d30e49a8f6c11ab3e44d91c/src/dtgtk/paint.h#L26-L44

Some of these describe behavior which overlap with what is now theme-able via CSS. At first glance, some which could be implemented via CSS:

- `CPF_ACTIVE`
- `CPF_PRELIGHT`
- `CPF_IGNORE_FG_STATE`
- `CPF_BG_TRANSPARENT`
- `CPF_STYLE_FLAT`
- `CPF_STYLE_BOX`
- `CPF_DO_NOT_USE_BORDER`
- `CPF_CUSTOM_BG`
- `CPF_CUSTOM_FG`

In addition, the current code to implement these flags isn't always intutive:

https://github.com/darktable-org/darktable/blob/ad87231986fa519d2d30e49a8f6c11ab3e44d91c/src/dtgtk/togglebutton.c#L101-L115

A CSS implementation may be more clear, and allow for more interesting design choices.

#### Plan A (lower impact)
Keep these flags as is (e.g. API for custom widget creation is unchanged). In the widget draw code, set a CSS class depending on the icon flags. Write appropriate CSS for each class to make behavior equivalent to what we have now.

#### Plan B (higher impact)
Change the API for widget creation to drop these flags. Still use CSS classes to describe the various behaviors. Each `gui_init()` routine which creates widgets will need to set assign the appropriate class to each widget to get the right visual behavior. Will produce more verbose C init code with assumptions built in about varieties of display. CSS will be fairly succinct.

#### Plan C (CSS-centric)
Change the API for widget creation to drop these flags. Rely entirely on CSS selectors to assign the appropriate style to the widgets. May produce more verbose CSS, but all styling can be done without altering the C code. OTOH, if there are no clear CSS classes, this could lead to a variety of unharmonious widget styles.

### Alternatives
Leave things as is. Down side is that I think everyone gets confused about what `CPF_STYLE_FLAT` means, and there are still workarounds in CSS for limitations of current styling. But won't require a lot of working and mucking around in widget creation routines and CSS to build something which will be visually indistinguishable from what exists now.

It's also possible to start with Plan A (above) and move on to Plan B/C later, to do the work in stages.

### Additional context
This came up when thinking out how to style widgets for #7952.

I'd be happy to work on this, though I also think it's not a high priority -- more to clean up existing overlapping code and allow for more flexible styling choices in the future.

### PR DESCRIPTION
**Pull request by AlicVB**
fix #7968

This is a big cleanup in buttons/paint flags.
Currently we have lot of drawing flags which are used to "theme" the buttons. Now that all this is managed by css, better remove them, that will avoid a mix between hardcoded things and css ;)

I think that it doesn't change anything in the UI, but better test it twice, considering the amount of changes done in so many files !

### CODE REVIEW

File: src/develop/blend_gui.c [hunk 1]
```diff
@@ -2328,8 +2327,8 @@ void dt_iop_gui_init_masks(GtkBox *blendw, dt_iop_module_t *module)
     bd->masks_polarity = dt_iop_togglebutton_new(module, "blend`tools", N_("toggle polarity of drawn mask"), NULL,
                                                  G_CALLBACK(_blendop_masks_polarity_callback),
                                                  FALSE, 0, 0, dtgtk_cairo_paint_plusminus, hbox);
-    dtgtk_togglebutton_set_paint(DTGTK_TOGGLEBUTTON(bd->masks_polarity), dtgtk_cairo_paint_plusminus,
-                                 CPF_STYLE_FLAT | CPF_BG_TRANSPARENT | CPF_IGNORE_FG_STATE, NULL);
+    dtgtk_togglebutton_set_paint(DTGTK_TOGGLEBUTTON(bd->masks_polarity), dtgtk_cairo_paint_plusminus, 0, NULL);
+    gtk_style_context_add_class(gtk_widget_get_style_context(bd->masks_polarity), "dt_ignore_fg_state");
```
Comment (Nilvus) [inline src/develop/blend_gui.c:2331]: Use dt_gui_add_class() function here

File: src/develop/blend_gui.c [hunk 2]
```diff
@@ -2177,8 +2176,8 @@ void dt_iop_gui_init_blendif(GtkBox *blendw, dt_iop_module_t *module)
                                                    in_out ? "blend-upper" : "blend-lower"));
       gtk_box_pack_start(GTK_BOX(slider_box), GTK_WIDGET(sl->slider), TRUE, TRUE, 0);
 
-      sl->polarity
-          = dtgtk_togglebutton_new(dtgtk_cairo_paint_plusminus, CPF_STYLE_FLAT | CPF_BG_TRANSPARENT | CPF_IGNORE_FG_STATE, NULL);
+      sl->polarity = dtgtk_togglebutton_new(dtgtk_cairo_paint_plusminus, 0, NULL);
+      gtk_style_context_add_class(gtk_widget_get_style_context(sl->polarity), "dt_ignore_fg_state");
```
Comment (Nilvus) [inline src/develop/blend_gui.c:2180]: Use dt_gui_add_class() function here

File: src/develop/blend_gui.c [hunk 3]
```diff
@@ -2340,26 +2339,31 @@ void dt_iop_gui_init_masks(GtkBox *blendw, dt_iop_module_t *module)
     bd->masks_shapes[0] = dt_iop_togglebutton_new(module, "blend`shapes", N_("add gradient"), N_("add multiple gradients"),
                                                   G_CALLBACK(_blendop_masks_add_shape),
                                                   FALSE, 0, 0, dtgtk_cairo_paint_masks_gradient, abox);
+    gtk_style_context_add_class(gtk_widget_get_style_context(bd->masks_shapes[0]), "dt_checked_background");
```
Comment (Nilvus) [inline src/develop/blend_gui.c:2342]: Use dt_gui_add_class() function here

File: src/develop/blend_gui.c [hunk 4]
```diff
@@ -2340,26 +2339,31 @@ void dt_iop_gui_init_masks(GtkBox *blendw, dt_iop_module_t *module)
...
 
     bd->masks_type[4] = DT_MASKS_BRUSH;
     bd->masks_shapes[4] = dt_iop_togglebutton_new(module, "blend`shapes", N_("add brush"), N_("add multiple brush strokes"),
                                                   G_CALLBACK(_blendop_masks_add_shape),
                                                   FALSE, 0, 0, dtgtk_cairo_paint_masks_brush, abox);
+    gtk_style_context_add_class(gtk_widget_get_style_context(bd->masks_shapes[4]), "dt_checked_background");
```
Comment (Nilvus) [inline src/develop/blend_gui.c:2348]: Use dt_gui_add_class() function here

File: src/develop/blend_gui.c [hunk 5]
```diff
@@ -2340,26 +2339,31 @@ void dt_iop_gui_init_masks(GtkBox *blendw, dt_iop_module_t *module)
...
 
     bd->masks_type[1] = DT_MASKS_PATH;
     bd->masks_shapes[1] = dt_iop_togglebutton_new(module, "blend`shapes", N_("add path"), N_("add multiple paths"),
                                                   G_CALLBACK(_blendop_masks_add_shape),
                                                   FALSE, 0, 0, dtgtk_cairo_paint_masks_path, abox);
+    gtk_style_context_add_class(gtk_widget_get_style_context(bd->masks_shapes[1]), "dt_checked_background");
```
Comment (Nilvus) [inline src/develop/blend_gui.c:2354]: Use dt_gui_add_class() function here

File: src/develop/blend_gui.c [hunk 6]
```diff
@@ -2340,26 +2339,31 @@ void dt_iop_gui_init_masks(GtkBox *blendw, dt_iop_module_t *module)
...
 
     bd->masks_type[2] = DT_MASKS_ELLIPSE;
     bd->masks_shapes[2] = dt_iop_togglebutton_new(module, "blend`shapes", N_("add ellipse"), N_("add multiple ellipses"),
                                                   G_CALLBACK(_blendop_masks_add_shape),
                                                   FALSE, 0, 0, dtgtk_cairo_paint_masks_ellipse, abox);
+    gtk_style_context_add_class(gtk_widget_get_style_context(bd->masks_shapes[2]), "dt_checked_background");
```
Comment (Nilvus) [inline src/develop/blend_gui.c:2360]: Use dt_gui_add_class() function here

File: src/develop/blend_gui.c [hunk 7]
```diff
@@ -2340,26 +2339,31 @@ void dt_iop_gui_init_masks(GtkBox *blendw, dt_iop_module_t *module)
...
 
     bd->masks_type[3] = DT_MASKS_CIRCLE;
     bd->masks_shapes[3] = dt_iop_togglebutton_new(module, "blend`shapes", N_("add circle"), N_("add multiple circles"),
                                                   G_CALLBACK(_blendop_masks_add_shape),
                                                   FALSE, 0, 0, dtgtk_cairo_paint_masks_circle, abox);
+    gtk_style_context_add_class(gtk_widget_get_style_context(bd->masks_shapes[3]), "dt_checked_background");
```
Comment (Nilvus) [inline src/develop/blend_gui.c:2366]: Use dt_gui_add_class() function here

File: src/develop/blend_gui.c [hunk 8]
```diff
@@ -2503,8 +2507,8 @@ void dt_iop_gui_init_raster(GtkBox *blendw, dt_iop_module_t *module)
     dt_bauhaus_combobox_add_populate_fct(bd->raster_combo, _raster_combo_populate);
     gtk_box_pack_start(GTK_BOX(hbox), bd->raster_combo, TRUE, TRUE, 0);
 
-    bd->raster_polarity = dtgtk_togglebutton_new(dtgtk_cairo_paint_plusminus, CPF_STYLE_FLAT | CPF_BG_TRANSPARENT | CPF_IGNORE_FG_STATE,
-                                                 NULL);
+    bd->raster_polarity = dtgtk_togglebutton_new(dtgtk_cairo_paint_plusminus, 0, NULL);
+    gtk_style_context_add_class(gtk_widget_get_style_context(bd->raster_polarity), "dt_ignore_fg_state");
```
Comment (Nilvus) [inline src/develop/blend_gui.c:2511]: Use dt_gui_add_class() function here

File: src/dtgtk/button.c [hunk 9]
```diff
@@ -131,7 +122,6 @@ GtkWidget *dtgtk_button_new(DTGTKCairoPaintIconFunc paint, gint paintflags, void
   button->canvas = gtk_drawing_area_new();
   gtk_container_add(GTK_CONTAINER(button), button->canvas);
   dt_gui_add_class(GTK_WIDGET(button), "dt_module_btn");
-  gtk_widget_set_name(GTK_WIDGET(button), "dt-button");
```
Comment (Nilvus) [inline src/dtgtk/button.c:134]: Do not remove it. Still needed on some places to allow different settings between dt-button and dt-toggle-button, like in blending tabs

File: src/dtgtk/togglebutton.c [hunk 10]
```diff
@@ -162,7 +140,6 @@ GtkWidget *dtgtk_togglebutton_new(DTGTKCairoPaintIconFunc paint, gint paintflags
   button->canvas = gtk_drawing_area_new();
   gtk_container_add(GTK_CONTAINER(button), button->canvas);
   dt_gui_add_class(GTK_WIDGET(button), "dt_module_btn");
-  gtk_widget_set_name(GTK_WIDGET(button), "dt-toggle-button");
```
Comment (Nilvus) [inline src/dtgtk/togglebutton.c:165]: Do not remove it. Still needed on some places to allow different settings between dt-button and dt-toggle-button, like in blending tabs

File: src/gui/gtk.c [hunk 11]
```diff
@@ -1361,7 +1361,9 @@ int dt_gui_gtk_init(dt_gui_gtk_t *gui)
   dt_control_change_cursor(GDK_LEFT_PTR);
 
   // create focus-peaking button
-  darktable.gui->focus_peaking_button = dtgtk_togglebutton_new(dtgtk_cairo_paint_focus_peaking, CPF_STYLE_FLAT, NULL);
+  darktable.gui->focus_peaking_button = dtgtk_togglebutton_new(dtgtk_cairo_paint_focus_peaking, 0, NULL);
+  gtk_style_context_add_class(gtk_widget_get_style_context(darktable.gui->focus_peaking_button),
```
Comment (Nilvus) [inline src/gui/gtk.c:1365]: Use dt_gui_add_class() here

File: src/iop/filmic.c [hunk 12]
```diff
@@ -1721,8 +1721,8 @@ void gui_init(dt_iop_module_t *self)
 
   GtkWidget *destdisp_head = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, DT_BAUHAUS_SPACE);
   GtkWidget *destdisp = dt_ui_section_label_new(_("destination/display"));
-  g->extra_toggle =
-    dtgtk_togglebutton_new(dtgtk_cairo_paint_solid_arrow, CPF_STYLE_BOX | CPF_DIRECTION_LEFT, NULL);
+  g->extra_toggle = dtgtk_togglebutton_new(dtgtk_cairo_paint_solid_arrow, CPF_DIRECTION_LEFT, NULL);
+  gtk_widget_set_name(GTK_WIDGET(g->extra_toggle), "control-button");
```
Comment (Nilvus) [inline src/iop/filmic.c:1725]: I remove it on purpose, not to see this add back. My CSS on last PR was fix that without need of it

File: src/libs/modulegroups.c [hunk 13]
```diff
@@ -639,9 +640,9 @@ static void _basics_add_widget(dt_lib_module_t *self, dt_lib_modulegroups_basic_
     if(!compact_ui)
     {
       // we add the on-off button
-      GtkWidget *btn
-          = dtgtk_togglebutton_new(dtgtk_cairo_paint_switch, CPF_STYLE_FLAT | CPF_BG_TRANSPARENT, item->module);
-      dt_gui_add_class(btn, "dt_module_btn");
+      GtkWidget *btn = dtgtk_togglebutton_new(dtgtk_cairo_paint_switch, 0, item->module);
+      gtk_widget_set_name(btn, "module-enable-button");
```
Comment (Nilvus) [inline src/libs/modulegroups.c:644]: Je l'avais enlevé celui-là

File: src/libs/modulegroups.c [hunk 14]
```diff
@@ -472,8 +472,9 @@ static void _basics_add_widget(dt_lib_module_t *self, dt_lib_modulegroups_basic_
 
       // we create a new button linked with the real one
       // because it create too much pb to remove the button from the expander
-      GtkWidget *btn
-          = dtgtk_togglebutton_new(dtgtk_cairo_paint_switch, CPF_STYLE_FLAT | CPF_BG_TRANSPARENT, item->module);
+      GtkWidget *btn = dtgtk_togglebutton_new(dtgtk_cairo_paint_switch, 0, item->module);
+      gtk_widget_set_name(btn, "module-enable-button");
```
Comment (Nilvus) [inline src/libs/modulegroups.c:476]: Je l'avais enlevé celui-là

File: src/libs/modulegroups.c [hunk 15]
```diff
@@ -2042,8 +2043,9 @@ static void _manage_editor_basics_update_list(dt_lib_module_t *self)
           gtk_box_pack_start(GTK_BOX(hb), lb, FALSE, TRUE, 0);
           if(!d->edit_ro)
           {
-            GtkWidget *btn = dtgtk_button_new(dtgtk_cairo_paint_cancel, CPF_STYLE_FLAT, NULL);
-            dt_gui_add_class(btn, "dt_module_btn");
+            GtkWidget *btn = dtgtk_button_new(dtgtk_cairo_paint_cancel, 0, NULL);
+            gtk_widget_set_name(btn, "module-reset-button");
```
Comment (Nilvus) [inline src/libs/modulegroups.c:2047]: Celui-là aussi

File: src/libs/modulegroups.c [hunk 16]
```diff
@@ -2818,7 +2818,8 @@ void gui_init(dt_lib_module_t *self)
   gtk_box_pack_start(GTK_BOX(d->hbox_buttons), d->hbox_groups, TRUE, TRUE, 0);
 
   // basic group button
-  d->basic_btn = dtgtk_togglebutton_new(dtgtk_cairo_paint_modulegroup_basics, pf, NULL);
+  d->basic_btn = dtgtk_togglebutton_new(dtgtk_cairo_paint_modulegroup_basics, 0, NULL);
+  gtk_style_context_add_class(gtk_widget_get_style_context(d->basic_btn), "dt_checked_background");
```
Comment (Nilvus) [inline src/libs/modulegroups.c:2822]: Use dt_gui_add_class() here

File: src/libs/modulegroups.c [hunk 17]
```diff
@@ -2828,14 +2829,15 @@ void gui_init(dt_lib_module_t *self)
   d->basics = NULL;
 
   // active group button
-  d->active_btn = dtgtk_togglebutton_new(dtgtk_cairo_paint_modulegroup_active, pf, NULL);
+  d->active_btn = dtgtk_togglebutton_new(dtgtk_cairo_paint_modulegroup_active, 0, NULL);
+  gtk_style_context_add_class(gtk_widget_get_style_context(d->active_btn), "dt_checked_background");
```
Comment (Nilvus) [inline src/libs/modulegroups.c:2833]: Same here

File: src/libs/modulegroups.c [hunk 18]
```diff
@@ -2961,7 +2963,8 @@ static void _buttons_update(dt_lib_module_t *self)
   for(l = d->groups; l; l = g_list_next(l))
   {
     dt_lib_modulegroups_group_t *gr = (dt_lib_modulegroups_group_t *)l->data;
-    GtkWidget *bt = dtgtk_togglebutton_new(_buttons_get_icon_fct(gr->icon), CPF_STYLE_FLAT, NULL);
+    GtkWidget *bt = dtgtk_togglebutton_new(_buttons_get_icon_fct(gr->icon), 0, NULL);
+    gtk_style_context_add_class(gtk_widget_get_style_context(bt), "dt_checked_background");
```
Comment (Nilvus) [inline src/libs/modulegroups.c:2967]: And here

File: src/libs/tools/global_toolbox.c [hunk 19]
```diff
@@ -399,7 +399,8 @@ void gui_init(dt_lib_module_t *self)
   self->widget = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0);
 
   /* create the grouping button */
-  d->grouping_button = dtgtk_togglebutton_new(dtgtk_cairo_paint_grouping, CPF_STYLE_FLAT, NULL);
+  d->grouping_button = dtgtk_togglebutton_new(dtgtk_cairo_paint_grouping, 0, NULL);
+  gtk_style_context_add_class(gtk_widget_get_style_context(d->grouping_button), "dt_checked_background");
```
Comment (Nilvus) [inline src/libs/tools/global_toolbox.c:403]: Use dt_gui_add_class() here

File: src/libs/tools/global_toolbox.c [hunk 20]
```diff
@@ -504,14 +505,16 @@ void gui_init(dt_lib_module_t *self)
   gtk_widget_show(vbox);
 
   /* create the widget help button */
-  d->help_button = dtgtk_togglebutton_new(dtgtk_cairo_paint_help, CPF_STYLE_FLAT, NULL);
+  d->help_button = dtgtk_togglebutton_new(dtgtk_cairo_paint_help, 0, NULL);
+  gtk_style_context_add_class(gtk_widget_get_style_context(d->help_button), "dt_checked_background");
```
Comment (Nilvus) [inline src/libs/tools/global_toolbox.c:509]: Same here

File: src/libs/tools/global_toolbox.c [hunk 21]
```diff
@@ -504,14 +505,16 @@ void gui_init(dt_lib_module_t *self)
...
   gtk_box_pack_start(GTK_BOX(self->widget), d->help_button, FALSE, FALSE, 0);
   gtk_widget_set_tooltip_text(d->help_button, _("enable this, then click on a control element to see its online help"));
   g_signal_connect(G_OBJECT(d->help_button), "clicked", G_CALLBACK(_lib_help_button_clicked), d);
   dt_gui_add_help_link(d->help_button, dt_get_help_url("global_toolbox_help"));
 
   /* create the shortcuts button */
-  d->keymap_button = dtgtk_togglebutton_new(dtgtk_cairo_paint_shortcut, CPF_STYLE_FLAT, NULL);
+  d->keymap_button = dtgtk_togglebutton_new(dtgtk_cairo_paint_shortcut, 0, NULL);
+  gtk_style_context_add_class(gtk_widget_get_style_context(d->keymap_button), "dt_checked_background");
```
Comment (Nilvus) [inline src/libs/tools/global_toolbox.c:517]: And here

File: src/views/darkroom.c [hunk 22]
```diff
@@ -2257,33 +2257,32 @@ void gui_init(dt_view_t *self)
...
   g_signal_connect(G_OBJECT(styles), "clicked", G_CALLBACK(_darkroom_ui_apply_style_popupmenu), NULL);
   gtk_widget_set_tooltip_text(styles, _("quick access for applying any of your styles"));
   dt_gui_add_help_link(styles, dt_get_help_url("bottom_panel_styles"));
   dt_view_manager_view_toolbox_add(darktable.view_manager, styles, DT_VIEW_DARKROOM);
 
   /* create second window display button */
-  dev->second_window.button
-      = dtgtk_togglebutton_new(dtgtk_cairo_paint_display2, CPF_STYLE_FLAT, NULL);
+  dev->second_window.button = dtgtk_togglebutton_new(dtgtk_cairo_paint_display2, 0, NULL);
+  gtk_style_context_add_class(gtk_widget_get_style_context(dev->second_window.button), "dt_checked_background");
```
Comment (Nilvus) [inline src/views/darkroom.c:2276]: Use dt_gui_add_class

File: src/views/darkroom.c [hunk 23]
```diff
@@ -2257,33 +2257,32 @@ void gui_init(dt_view_t *self)
...
   g_signal_connect(G_OBJECT(dev->second_window.button), "clicked", G_CALLBACK(_second_window_quickbutton_clicked),
                    dev);
   gtk_widget_set_tooltip_text(dev->second_window.button, _("display a second darkroom image window"));
   dt_view_manager_view_toolbox_add(darktable.view_manager, dev->second_window.button, DT_VIEW_DARKROOM);
 
   /* Enable ISO 12646-compliant colour assessment conditions */
-  dev->iso_12646.button
-      = dtgtk_togglebutton_new(dtgtk_cairo_paint_bulb, CPF_STYLE_FLAT, NULL);
+  dev->iso_12646.button = dtgtk_togglebutton_new(dtgtk_cairo_paint_bulb, 0, NULL);
+  gtk_style_context_add_class(gtk_widget_get_style_context(dev->iso_12646.button), "dt_checked_background");
```
Comment (Nilvus) [inline src/views/darkroom.c:2285]: Same here

File: src/views/darkroom.c [hunk 24]
```diff
@@ -2295,8 +2294,8 @@ void gui_init(dt_view_t *self)
   /* create rawoverexposed popup tool */
   {
     // the button
-    dev->rawoverexposed.button
-        = dtgtk_togglebutton_new(dtgtk_cairo_paint_rawoverexposed, CPF_STYLE_FLAT, NULL);
+    dev->rawoverexposed.button = dtgtk_togglebutton_new(dtgtk_cairo_paint_rawoverexposed, 0, NULL);
+    gtk_style_context_add_class(gtk_widget_get_style_context(dev->rawoverexposed.button), "dt_checked_background");
```
Comment (Nilvus) [inline src/views/darkroom.c:2298]: And here

File: src/views/darkroom.c [hunk 25]
```diff
@@ -2350,8 +2349,8 @@ void gui_init(dt_view_t *self)
   /* create overexposed popup tool */
   {
     // the button
-    dev->overexposed.button
-        = dtgtk_togglebutton_new(dtgtk_cairo_paint_overexposed, CPF_STYLE_FLAT, NULL);
+    dev->overexposed.button = dtgtk_togglebutton_new(dtgtk_cairo_paint_overexposed, 0, NULL);
+    gtk_style_context_add_class(gtk_widget_get_style_context(dev->overexposed.button), "dt_checked_background");
```
Comment (Nilvus) [inline src/views/darkroom.c:2353]: same here

File: src/views/darkroom.c [hunk 26]
```diff
@@ -2416,8 +2415,9 @@ void gui_init(dt_view_t *self)
   /* create profile popup tool & buttons (softproof + gamut) */
   {
     // the softproof button
-    dev->profile.softproof_button =
-      dtgtk_togglebutton_new(dtgtk_cairo_paint_softproof, CPF_STYLE_FLAT, NULL);
+    dev->profile.softproof_button = dtgtk_togglebutton_new(dtgtk_cairo_paint_softproof, 0, NULL);
+    gtk_style_context_add_class(gtk_widget_get_style_context(dev->profile.softproof_button),
```
Comment (Nilvus) [inline src/views/darkroom.c:2419]: And here

File: src/views/darkroom.c [hunk 27]
```diff
@@ -2427,8 +2427,8 @@ void gui_init(dt_view_t *self)
     dt_gui_add_help_link(dev->profile.softproof_button, dt_get_help_url("softproof"));
 
     // the gamut check button
-    dev->profile.gamut_button =
-      dtgtk_togglebutton_new(dtgtk_cairo_paint_gamut_check, CPF_STYLE_FLAT, NULL);
+    dev->profile.gamut_button = dtgtk_togglebutton_new(dtgtk_cairo_paint_gamut_check, 0, NULL);
+    gtk_style_context_add_class(gtk_widget_get_style_context(dev->profile.gamut_button), "dt_checked_background");
```
Comment (Nilvus) [inline src/views/darkroom.c:2431]: ANd here

File: src/views/darkroom.c [hunk 28]
```diff
@@ -2590,7 +2590,9 @@ void gui_init(dt_view_t *self)
   /* create grid changer popup tool */
   {
     // the button
-    darktable.view_manager->guides_toggle = dtgtk_togglebutton_new(dtgtk_cairo_paint_grid, CPF_STYLE_FLAT, NULL);
+    darktable.view_manager->guides_toggle = dtgtk_togglebutton_new(dtgtk_cairo_paint_grid, 0, NULL);
+    gtk_style_context_add_class(gtk_widget_get_style_context(darktable.view_manager->guides_toggle),
```
Comment (Nilvus) [inline src/views/darkroom.c:2594]: Here too

**Review by Nilvus — CHANGES_REQUESTED** [review]
Many changes are needed here and I didn't review darktable.css. I let some #dt-ubtton and #dt-toggle-button on purpose. Cleanup is really good but go too far here. I'm even wondering if the rebase is done completely on master.

**Comment by Nilvus** [comment]
Just a quick code review. I will test more tomorrow but many more things on CSS need to be update and for some revert to what I've done. But even on dt-button, and dt-toggle-button, maybe some of your code is good.

File: data/themes/darktable.css [hunk 29]
```diff
@@ -723,7 +750,7 @@ spinbutton>button
 }
 
 /* set background color and color of scope/histogram buttons */
-#main-histogram #dt-button
+#main-histogram .dt_module_btn
```
Comment (Nilvus) [inline data/themes/darktable.css:753]: This break histogram button and especially hover effect. Just check them on histogram and see hover effect on master and on your PR.

File: data/themes/darktable.css [hunk 30]
```diff
@@ -1016,33 +1043,33 @@ notebook tabs,
...
+#guides-line .dt_module_btn
 {
   min-width: 1.5em;
 }
 
 /* reduce and align now hamburger menu of modules and blending tabs */
-#blending-tabs #dt-button,
-#modules-tabs #dt-button
+#blending-tabs .dt_module_btn,
+#modules-tabs .dt_module_btn
```
Comment (Nilvus) [inline data/themes/darktable.css:1067]: This make hamburger menu too big on both parts. With lines just above, it's needed and wanted to keep difference between dt-toggle-button and dt-button

File: data/themes/darktable.css [hunk 31]
```diff
@@ -1016,33 +1043,33 @@ notebook tabs,
...
   padding: 0.28em;
 }
 
-#modules-tabs #dt-toggle-button,   /* This is needed separately to set a minimal width if panel is really narrow */
-#blending-tabs #dt-toggle-button,
-#guides-line #dt-toggle-button
+#modules-tabs .dt_module_btn,   /* This is needed separately to set a minimal width if panel is really narrow */
+#blending-tabs .dt_module_btn,
+#guides-line .dt_module_btn
 {
```
Comment (Nilvus) [inline data/themes/darktable.css:1061]: Revert back to dt-toggle-button here (see comment below)

File: data/themes/darktable.css [hunk 32]
```diff
@@ -1016,33 +1043,33 @@ notebook tabs,
...
-#blending-tabs #dt-button,
-#modules-tabs #dt-button
+#blending-tabs .dt_module_btn,
+#modules-tabs .dt_module_btn
 {
   padding: 0.2em 0.5em;
 }
 
-#blending-tabs #dt-button
+#blending-tabs .dt_module_btn
```
Comment (Nilvus) [inline data/themes/darktable.css:1072]: Needed to put back dt-button here (for same reason as above

**Review by Nilvus — CHANGES_REQUESTED** [review]
Some first request on darktable.css files. Probably more to come

File: data/themes/darktable.css (hunk 29 above)
Comment (Nilvus) [reply data/themes/darktable.css:753]: Ignore that comment

**Review by Nilvus — APPROVED** [review]
All good now. Thanks for this great work.

File: src/libs/modulegroups.c [hunk 33]
```diff
@@ -2193,7 +2192,7 @@ static void _manage_editor_module_update_list(dt_lib_module_t *self, dt_lib_modu
         gtk_box_pack_start(GTK_BOX(hb), lb, FALSE, TRUE, 0);
         if(!d->edit_ro)
         {
-          GtkWidget *btn = dtgtk_button_new(dtgtk_cairo_paint_cancel, CPF_STYLE_FLAT, NULL);
+          GtkWidget *btn = dtgtk_button_new(dtgtk_cairo_paint_cancel, 0, NULL);
           gtk_widget_set_name(btn, "module-reset-button");
```
Comment (Nilvus) [inline src/libs/modulegroups.c:2196]: Oh just see that one to remove too!

**Review by Nilvus — CHANGES_REQUESTED** [review]

**Review by Nilvus — APPROVED** [review]

**Comment by Nilvus** [comment]
Ready to be merged!

**Review by elstoc — APPROVED** [review]
Looks good. I don't see any obvious regressions

**Comment by AlicVB** [comment]
note that at some point, we'll also have to do a big cleanup in how CPF_... flags are defined : 
- reordering them by putting state flags in first as their number are limited (eventually let 1-2 places for eventual new state flags)
- fixing all the places where flags values are called by their int values directly

but I would prefer to do that after the filter PR as land, to avoid big headaches in rebasing... btw, I'll certainly need your help @phweyland as you are doing some strange things with this in your colorfilter... I'll ping you ;)

**Comment by phweyland** [comment]
> you are doing some strange things with this in your colorfilter.

Please tell me where, I'll try to understand what I've done ...

**Comment by AlicVB** [comment]
this is in src/lib/tools/filters.c for example in `_update_colors_filter`...
Note that I don't say it's wrong or whatever (I'm sure of the contrary)... just that I don't understand what is done exactly and so I fear to broke something if I change the numeric value of CPF flags...

Anyway, as said, as this part is moved in another place in code with my filtering PR, I would prefer to do this work once the PR merged...

**Review by Nilvus — APPROVED** [review]

**Review by TurboGit — APPROVED** [review]
Works for me, thanks!

**Pull request merged**